Sources/Sentry/Public/SentrySpan.h

```diff
@@ -1,9 +1,97 @@
-#import <Foundation/Foundation.h>
+#import "SentrySpanContext.h"
+#import <Sentry/Sentry.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
+@class SentryTransaction, SentrySpanId, SentryId, SentryHub;
+
 NS_SWIFT_NAME(Span)
-@protocol SentrySpan <NSObject>
+@interface SentrySpan : SentrySpanContext
+SENTRY_NO_INIT
+
+/**
+ * The timestamp of which the span ended.
+ */
+@property (nullable, nonatomic, strong) NSDate *timestamp;
+
+/**
+ * The start time of the span.
+ */
+@property (nullable, nonatomic, strong) NSDate *startTimestamp;
+
+/**
+ * An arbitrary mapping of additional metadata of the span.
+ */
+@property (nullable, readonly) NSDictionary<NSString *, id> *extras;
```

Comment:
.NET called it Extras. In documentation is "data" (https://develop.sentry.dev/sdk/event-payloads/span/). Java does not have it.

Reply:
mmm wondering why we don't have it on Java, I'll check that.

Reply:
https://github.com/getsentry/sentry-java/issues/1233

Reply:
@bruno-garcia what's about using `data` instead of `extras`, the reason is to not get confused with Scope.extra

Reply:
yeah makes sense to align, no strong opinion on each way to go, data sounds good

Reply:
@Tyrrrz

Reply:
Yes, it's the `data` property. On .NET it's called `extra` to be consistent with other models that have a similar string to object key-value store. When serialized to JSON, it's mapped as `data`.

Reply:
As far as I understood originally, both `data` in spans and `extra` everywhere else are functionally the same, just named differently. Or are there some differences?

Reply:
the `transaction` does not have a `data` field in the protocol, but `extra` and it's semantically the same as Span#data yes.

it's just weird to set the same thing with different methods eg

```
transacion.setExtra(key, value)
span.setData(key, value)
```

so I'd propose to unify this setting, right now on Java, we'll do like this:

```
transacion.setData(key, value) // forwards to transacion.setExtra(key, value), JSON as `extra`
span.setData(key, value) // JSON as `data`
```

Reply:
@Tyrrrz maybe if not worth adding the method forwarders called `setData`, let's just write on method docs that its the equivalent of the field `data` in the protocol?

Reply:
I would rather stick with `extra` as we already use it already on the event and the scope. So I would say it should be `setExtra` for both transaction and span.

Reply:
I also think consolidating on `setExtra` makes more sense, since we already use it everywhere else besides spans.

Reply:
the thing is, `transaction.data` goes under `additional data` which is the equivalent of `event.extra` as `transaction.data` does not exist yet, on the server.

`span.data` goes under the transaction details along with other information.

JS and Python expose `Span#setData` and `Transaction#setExtra`, my idea was to keep the same behavior, but adding a method `Transaction#setData` that forwards to `Transaction#setExtra`, so the Transaction/Span would be easier to use, even though `transaction.data` would be parsed as `extra` anyway.

Right now every SDK is different, the idea is to have a similar API, I don't mind either one or the other, but right now we have 4 variations and this is suboptimal.

Reply:
@bruno-garcia opinions?

Reply:
> the thing is, transaction.data goes under additional data which is the equivalent of event.extra as transaction.data does not exist yet, on the server.

`data` doesnt' exist and won't be added to the protocol.

transaction is event and we made stuff up here by splitting it. span.data is likely what other SDKs added, but transaction.data doesn't make sense though

Reply:
I would prefer we don't add Transaction.setData. It'll diverge in documentation from other SDKs, and setExtra will be there anyway since it's part of the event.

We seem to have lots of opinions about this, maybe we discuss on a call?

Reply:
> Right now every SDK is different, the idea is to have a similar API, I don't mind either one or the other, but right now we have 4 variations and this is suboptimal.

I understand and agree with this.

Reply:
this is part of our agenda tomorrow

Reply:
Since Transaction will not be expose, this is kind solved, I guess.